add monorepo blogpost (#2606)
nf-core/website: sites/main-site/src/content/blog/2024/2024-07-02-new-website-structure.mdx

We accomplished this by keeping them in `sites/main-site/src` and setting aliases for the relative links
to `sites/main-site` in the respective sub-sites. This kept the actual needed code changes to a minimum.
Even though each of these sub-sites are their own entity on our host Netlify, they are all reachable under
the same `nf-co.re` domain. This is achieved by setting up a redirect rules with a `200` redirect status,
which is a silent redirect, meaning the URL in the browser does not change.

Data flow for a request to the homepage:

```mermaid
flowchart LR
%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
    A[fas:fa-link https://nf-co.re] --> B(fas:fa-server sites/main-site)
```

Data flow between sub-sites for a request to a pipeline page:

```mermaid
flowchart LR
%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%

        E(fas:fa-server sites/main-site)  --fetching components--> D(fas:fa-server sites/pipelines)

    C[fas:fa-link https://nf-co.re/rnaseq/latest/docs/usage] --> B(fas:fa-server sites/main-site)  --redirects silently to --> D
```

# Changes to the development workflow

All npm commands now should specify the workspace they are run with.

> Instead of running the development server with `npm run dev{:bash}` you need to now run it for a specific sub-site, for example:

Reviewer: ```suggestion
Instead of running the development server with `npm run dev{:bash}` you need to now run it for a specific sub-site.

For example, to see the changes to a blog entry, you should now run:
```

```bash
npm run dev -w sites/main-sites
```

> if you want to see the changes to a blog entry. Be aware that pages not part of these sub-site will throw a 404 error.

Reviewer: ```suggestion
:::warn
Be aware that pages not part of these sub-site will throw a 404 error.
This only happens when previewing a single sub-site and should not happen on the final built website.
:::
```

# Additional Changes

> We couldn't update to Astro 4 with the old website structure, because it increased the build time to > 20 minutes leading to a timeout on netlify. With the new structure, we are now able to use newer Astro versions and benefit from the new features and optimizations, including [content collection caching](https://astro.build/blog/astro-350/#content-collections-build-cache-experimental), decreasing the build time for repeated builds even more.

Reviewer: Cue "better faster stronger" gif or similar 😅

---

> title: new website structure

Reviewer: ```suggestion
title: Faster nf-core website builds with a new back-end setup
```

Reviewer: Don't love this, but wonder if the original could be tweaked to give a bit more nuance. Given that most people won't see any difference in structure (eg. navigation).

Author: Will compromise on "Faster nf-core website builds with a new code setup" 🙂

> subtitle: 'divide et impera'

Reviewer: ```suggestion
subtitle: Divide et impera
```

-   [pipelines](https://github.com/nf-core/website/tree/main/sites/pipelines)
-   [modules and subworkflows](https://github.com/nf-core/website/tree/main/sites/modules-subworkflows)
-   [documentation](https://github.com/nf-core/website/tree/main/sites/docs)
-   [configs](https://github.com/nf-core/website/tree/main/sites/configs)

These are hosted in separate Netlify deployments which are stitched together with redirect rules. To the end user the experience is the same, but the build steps can now be handled independently from one another.

Finally, we separated the [pipeline results](https://github.com/nf-core/website/tree/main/sites/pipeline-results) pages, which are server-side rendered per request instead of being generated during the build step like the rest of the pages.

<Image
    src={subsites}
    class="d-block m-auto mb-2"
    width={400}
    density={[1.5, 2]}
    alt="Three headed dragon meme cartoon, with the title 'nf-co.re sub-sites'. The left head looks angry and has the title 'pipelines', the second head has the title 'docs' and looks stern at the third head, which looks very goofy and has the title 'pipeline-results'."
/>

# Results: build fast

> The results speak for themselves. This plot shows the build times for each sub-site, compared to the old site structure.:

Reviewer: ```suggestion
The results speak for themselves. This plot shows the build times for each sub-site, compared to the old site structure:
```
